eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageTypes.scala

```diff
@@ -253,21 +252,41 @@ case class ChannelUpdate(signature: ByteVector64,
                          chainHash: ByteVector32,
                          shortChannelId: ShortChannelId,
                          timestamp: Long,
-                         messageFlags: Byte,
-                         channelFlags: Byte,
+                         channelFlags: ChannelUpdate.ChannelFlags,
                          cltvExpiryDelta: CltvExpiryDelta,
                          htlcMinimumMsat: MilliSatoshi,
                          feeBaseMsat: MilliSatoshi,
                          feeProportionalMillionths: Long,
                          htlcMaximumMsat: Option[MilliSatoshi],
                          tlvStream: TlvStream[ChannelUpdateTlv] = TlvStream.empty) extends RoutingMessage with AnnouncementMessage with HasTimestamp with HasChainHash {
-  require(((messageFlags & 1) != 0) == htlcMaximumMsat.isDefined, "htlcMaximumMsat is not consistent with messageFlags")
 
-  def isNode1 = Announcements.isNode1(channelFlags)
+  def messageFlags: Byte = if (htlcMaximumMsat.isDefined) 1 else 0
 
   def toStringShort: String = s"cltvExpiryDelta=$cltvExpiryDelta,feeBase=$feeBaseMsat,feeProportionalMillionths=$feeProportionalMillionths"
 }
 
+object ChannelUpdate {
+  case class ChannelFlags(disable: Boolean, direction: Boolean) {
```

Comment:
Those booleans could have been flattened directly in the `ChannelUpdate`, that would have generated less boiler plate but was a bit less close to the "spirit of the spec". Let me know what you think!

Reply:
I agree, it's better to keep the "spirit of the spec" and keep the codec easy to read, so that we immediately see whether it correctly matches the spec or not.

I find it slightly weird though that `direction` is a `Boolean`...

Reply:
Now that I think of it, even if we stick to 'spirit of the spec" we have no reason to keep the same types and names in our classes. That's what we do everywhere else, but since those are booleans, weirdly we kept them as-is, and defined redundant more-meaningful methods to interpret them. We could (should I think) define `ChannelFlags` as:

```scala
case class ChannelFlags(isEnabled: Boolean, isNode1: Boolean)
```

Reply:
Done in 130f44772.